app/controllers/provider_interface/candidates_controller.rb

```diff
@@ -0,0 +1,37 @@
+module ProviderInterface
+  class CandidatesController < ProviderInterfaceController
+    before_action :disable_on_production, only: :impersonate
+
+    def impersonate
+      candidate = Candidate.find(params[:candidate_id])
+
+      if verify_provider_association(candidate: candidate, providers: current_provider_user.providers)
+        bypass_sign_in(candidate, scope: :candidate)
+
+        flash[:success] = "You are now signed in as candidate #{candidate.email_address}"
+
+        redirect_to candidate_interface_application_form_path
+      else
+        flash[:warning] = 'Candidate is not associated with your providers'
```

Comment:
They might only have one provider. How about "This candidate does not have applications you can manage" or equiv?

Reply:
Actually, how about just 'This candidate is not associated with your account'? It doesn't really matter, as you'll probably never see this message. This action is not possible through the UI (you're browsing applications already associated with your account) and to post to an arbitrary ID you'd have to be creative (perhaps use a different tool, play with the authenticity token etc.) So, you deserve a badly-worded error message!

Reply:
Nice :+1: